### PR TITLE
filter select column dropdown

### PR DESCRIPTION
**Pull request by leoraba**
- Adding a text filter to dropdown component.
- Prettify code

### CODE REVIEW

File: modules/components/src/DataTable/DataTable.js [hunk 1]
```diff
@@ -9,194 +9,196 @@ import noopFn from '@/utils/noops';
...
+			apiUrl = ARRANGER_API,
+			columnDropdownText,
+			config,
+			customActions = null,
+			customHeaderContent = null,
+			data = null,
+			documentType,
+			downloadUrl = '',
+			enableDropDownControls = false,
+			enableSelectColumnDropdownTextFilter = false,
```
Comment (leoraba) [inline modules/components/src/DataTable/DataTable.js:97]: Adding a new property, defaults to false.

File: modules/components/src/DropDown/DropDown.js [hunk 2]
```diff
@@ -5,135 +5,160 @@ import { css } from '@emotion/react';
...
+										placeholder={searchPlaceholder}
+										type="text"
+										value={searchText}
+									/>
+								)}
+								{items
+									.filter(
+										(item) =>
+											!searchText ||
+											internalTranslateSQONValue(itemToString(item)).match(strToReg(searchText)),
```
Comment (leoraba) [inline modules/components/src/DropDown/DropDown.js:119]: filter matching text.

File: modules/components/src/DropDown/DropDown.js [hunk 3]
```diff
@@ -5,135 +5,160 @@ import { css } from '@emotion/react';
...
+										return (
+											<div
+												className={`dropDownContentElement${
+													labelIsComponent ? ' custom' : ' clickable'
+												}`}
+												key={item.id || id}
+												{...itemProps}
+											>
+												{enableSelectColumnDropdownTextFilter ? (
+													<TextHighlight content={label} highlightText={searchText} />
```
Comment (leoraba) [inline modules/components/src/DropDown/DropDown.js:138]: using existing component to highlight matching text

File: modules/components/src/DropDown/DropDown.js [hunk 4]
```diff
@@ -5,135 +5,160 @@ import { css } from '@emotion/react';
...
+								style={{
+									right: align === 'right' ? 0 : 'auto',
+									left: align === 'right' ? 'auto' : 0,
+								}}
+								{...(singleSelect && { onClick: this.handleToggleMenu })}
+							>
+								{enableSelectColumnDropdownTextFilter && (
+									<TextFilter
+										aria-label={`Search data`}
+										onChange={({ value }) => this.setState({ searchText: value || '' })}
```
Comment (caravinci) [inline modules/components/src/DropDown/DropDown.js:109]: Think I'd abstract this function out of the return, so a new function is not instantiated with every render.

e.g. `onChange={handleFilter}`

File: modules/components/src/DropDown/DropDown.js [hunk 5]
```diff
@@ -5,135 +5,160 @@ import { css } from '@emotion/react';
...
+										placeholder={searchPlaceholder}
+										type="text"
+										value={searchText}
+									/>
+								)}
+								{items
+									.filter(
+										(item) =>
+											!searchText ||
```
Comment (caravinci) [inline modules/components/src/DropDown/DropDown.js:118]:
```suggestion
											// Filters out values that don't match the TextFilter's input
											!searchText ||
```

File: modules/components/src/DataTable/TableToolbar/TableToolbar.js [hunk 6]
```diff
@@ -123,175 +124,176 @@ const TableToolbar = ({
...
+							onChange={(item) => {
+								setFilterVal('');
+								onFilterChange({
+									value: '',
+									generateNextSQON: generateNextSQON(''),
+								});
+								onColumnsChange({ ...item, show: !item.show });
+							}}
+							theme={themeDropDownProps}
+							enableSelectColumnDropdownTextFilter={enableSelectColumnDropdownTextFilter}
```
Comment (caravinci) [inline modules/components/src/DataTable/TableToolbar/TableToolbar.js:201]:
```suggestion
							aria-label={`Select columns`}
							enableSelectColumnDropdownTextFilter={enableSelectColumnDropdownTextFilter}
							itemToString={(i) => i.displayName}
							items={canChangeShowColumns}
							onChange={(item) => {
								setFilterVal('');
								onFilterChange({
									value: '',
									generateNextSQON: generateNextSQON(''),
								});
								onColumnsChange({ ...item, show: !item.show });
							}}
							theme={themeDropDownProps}
```

To keep things consistent, we aim to have props alphabetised 💪

File: modules/components/src/DataTable/TableToolbar/TableToolbar.js [hunk 7]
```diff
@@ -123,175 +124,176 @@ const TableToolbar = ({
...
-          ))}
+			<div className="group">
+				{allowTogglingColumns &&
+					(enableDropDownControls ? (
+						<MultiSelectDropDown
+							allowControls
+							allowSelection
+							buttonAriaLabelClosed={`Open column selection menu`}
+							buttonAriaLabelOpen={`Close column selection menu`}
+							itemSelectionLegend={`Select columns to display`}
```
Comment (caravinci) [inline modules/components/src/DataTable/TableToolbar/TableToolbar.js:166]:
```suggestion
							enableSelectColumnDropdownTextFilter={enableSelectColumnDropdownTextFilter} // feature not implemented yet for this component
							itemSelectionLegend={`Select columns to display`}
```

**Review by caravinci — APPROVED** [review]
great work, @leoraba! 🙏 
Truly appreciate that you picked up the habit of leaving comments to guide through the PR.
left a couple change suggestions, and I'd hope the multiselect dropdown will get the same love at some point?

File: modules/components/src/DropDown/DropDown.js (hunk 4 above)
Comment (leoraba) [reply modules/components/src/DropDown/DropDown.js]: thanks for the suggestion! code refactored as suggested.

File: modules/components/src/DataTable/TableToolbar/TableToolbar.js (hunk 6 above)
Comment (leoraba) [reply modules/components/src/DataTable/TableToolbar/TableToolbar.js]: code changes as suggested

Comment (caravinci) [reply modules/components/src/DataTable/TableToolbar/TableToolbar.js]: awesome!
lol but remind me to show you how to accept commit suggestions (mine there was mergeable into your pr)

**Pull request closed without merging**